app/controllers/users/sessions_controller.rb

```diff
@@ -243,6 +243,10 @@ def override_csp_for_google_analytics
     def update_devise_params_sanitizer
       devise_parameter_sanitizer.permit(:sign_in, except: [:email, :password]) if !request.post?
     end
+
+    def sign_in_params
+      params[resource_name]&.permit(:email) if request.post?
+    end
```

Comment:
should we combine these two methods?

maybe something like:
```suggestion
    def update_devise_params_sanitizer
      if request.post?
        devise_parameter_sanitizer.permit(:sign_in, keys: [:email])
      else
        devise_parameter_sanitizer.permit(:sign_in, except: [:email, :password])
      end
    end

    def sign_in_params
      params[resource_name]&.permit(:email) if request.post?
    end
```

Reply:
The tests failed with this change, but everything appears to work with only changing `sign_in_params`:

```suggestion
    def sign_in_params
      params[resource_name]&.permit(:email) if request.post?
    end
```

Reply:
anything we can do to simplify, I'm a fan of! if we can remove that before_filter from both, great!

Reply:
Sounds good, applied that change and removed the `before_filter`